### ISSUE TITLE
onedb fsck error

### ISSUE THREAD

**Issue by FlorianHeigl** [OpenNebula/one#2248]
# Bug Report
## Version of OpenNebula

- [x] 5.2.2 (actually it's Nodeweaver, but I think this issue is mostly within ONE)
- [ ] 5.4.0
- [ ] 5.4.1
- [ ] 5.4.2
- [ ] 5.4.3
- [ ] 5.4.4
- [ ] 5.4.5
- [ ] 5.4.6
- [ ] 5.4.7
- [ ] 5.4.8
- [ ] 5.4.9
- [ ] 5.4.10
- [ ] 5.4.11
- [ ] 5.4.12
- [ ] 5.4.13
- [ ] 5.4.14
- [ ] 5.4.15
- [ ] Development build

## Component
- [ ] Authorization (LDAP, x509 certs...)
- [x] Command Line Interface (CLI)
- [ ] Contextualization
- [ ] Documentation
- [ ] Federation and HA
- [ ] Host, Clusters and Monitorization
- [ ] KVM
- [ ] Networking
- [ ] Orchestration (OpenNebula Flow)
- [ ] Packages
- [x] Scheduler
- [ ] Storage & Images
- [ ] Sunstone
- [ ] Upgrades
- [ ] User, Groups, VDCs and ACL
- [ ] vCenter

## Description

the scheduler is toast, and oned is not working well. tried to fsck, and am getting an error during onedb fsck.

```
$ onedb fsck -s one.db
Database schema does not look to be created by OpenNebula: table user_pool is missing or empty.
$ echo $?
255
```

I looked at the source and this code seems to be unchanged in master branch.
the message comes from one_db_backend.rb:

which contains this code 

```
                raise e
            elsif !db_exists?
                # If the DB doesn't have db_version table, it means it is empty or a 2.x
                raise "Database schema does not look to be created by " <<
                      "OpenNebula: table user_pool is missing or empty."
            end
```

which seems to be an incorrect message - there seems not be a specific check for user_pool, but only "db_exists" which seems to do something more, and not just look for user_pool (which _does_ exist and _has_ content). Then the fsck aborts and I get to stay with the corrupted database.

### Expected Behavior

Correct error message saying whats really wrong OR code that does whatever test is needed to identify if the user pool table is there / as expected.
Be able to run fsck.

### Actual Behavior

Messed up cluster noone can use ;-)

## How to reproduce

not sure. I can send the db.
btw maybe this all originates by nothing more than launching and terminating VMs quickly, I managed to see a deadlock on one.db. stopped oned, dump-restored the DB to get rid of the lock.

(fragile)

tl;dr:
ondb fsck will report incorrectly for most DB issues, the text is wrong, or the test is wrong.
